service/internal/access/v2/just_in_time_pdp.go

```diff
@@ -0,0 +1,333 @@
+package access
+
+import (
+	"context"
+	"errors"
+	"fmt"
+	"log/slog"
+
+	"github.com/opentdf/platform/lib/flattening"
+	authzV2 "github.com/opentdf/platform/protocol/go/authorization/v2"
+	"github.com/opentdf/platform/protocol/go/common"
+	"github.com/opentdf/platform/protocol/go/entity"
+	entityresolutionV2 "github.com/opentdf/platform/protocol/go/entityresolution/v2"
+	"github.com/opentdf/platform/protocol/go/policy"
+	attrs "github.com/opentdf/platform/protocol/go/policy/attributes"
+	"github.com/opentdf/platform/protocol/go/policy/subjectmapping"
+	otdfSDK "github.com/opentdf/platform/sdk"
+
+	"github.com/opentdf/platform/service/logger"
+)
+
+type JustInTimePDP struct {
+	logger *logger.Logger
+	sdk    *otdfSDK.SDK
+	// embedded PDP
+	pdp *PolicyDecisionPoint
+}
+
+// JustInTimePDP creates a new Policy Decision Point instance with no in-memory policy and a remote connection
+// via authenticated SDK, then fetches all Attributes and Subject Mappings from the policy services.
+func NewJustInTimePDP(
+	ctx context.Context,
+	l *logger.Logger,
+	sdk *otdfSDK.SDK,
+) (*JustInTimePDP, error) {
+	var err error
+
+	if sdk == nil {
+		l.ErrorContext(ctx, "invalid arguments", slog.String("error", ErrMissingRequiredSDK.Error()))
+		return nil, ErrMissingRequiredSDK
+	}
+	if l == nil {
+		l, err = logger.NewLogger(defaultFallbackLoggerConfig)
+		if err != nil {
+			return nil, fmt.Errorf("failed to initialize new PDP logger and none was provided: %w", err)
+		}
+	}
+
+	p := &JustInTimePDP{
+		sdk:    sdk,
+		logger: l,
+	}
+
+	allAttributes, err := p.fetchAllDefinitions(ctx)
+	if err != nil {
+		l.ErrorContext(ctx, "failed to fetch all attribute definitions", slog.String("error", err.Error()))
```

Comment:
Again, either do not log these or log at info or below to avoid duplicate error alerts in datadog/whatever